packages/lib/src/select.ts

```diff
@@ -4,32 +4,48 @@ import { CommonListOutputProducer } from './format'
 import { SmartThingsCommandInterface } from './smartthings-command'
 
 
-// Functions in this file support commands that act on an item. The biggest difference
-// between these and the methods from listing-io.ts is that they list items and immediately
-// query the user for an item to act on (if one wasn't specified on the command line). This
-// makes them safe to use for actions that make changes to the item or delete it.
+export type SelectingConfig<L> = Sorting & Naming & CommonListOutputProducer<L>
 
-// TODO: implement equivalent of acceptIndexId from old code
+export const indefiniteArticleFor = (name: string): string => name.match(/^[aeiou]/i) ? 'an' : 'a'
 
-export type SelectingConfig<L> = Sorting & Naming & CommonListOutputProducer<L>
+function promptFromNaming(config: Naming): string | undefined {
+	return config.itemName ? `Select ${indefiniteArticleFor(config.itemName)} ${config.itemName}.` : undefined
+}
 
 export async function selectGeneric<ID, L>(command: SmartThingsCommandInterface, config: SelectingConfig<L>,
 		preselectedId: ID | undefined, listItems: ListDataFunction<L>,
-		getIdFromUser: IdRetrievalFunction<ID, L>, promptMessage?: string): Promise<ID> {
+		getIdFromUser: IdRetrievalFunction<ID, L>, promptMessage?: string, autoChoose = false): Promise<ID> {
 	if (preselectedId) {
 		return preselectedId
 	}
 
-	const list = await outputList(command, config, listItems, true)
+	const items = await listItems()
+	if (autoChoose && items.length === 1) {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
```

Comment:
It would be helpful to mention what we know we are ignoring in the comment.

Reply:
Updated ignore comment.